src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,84 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int GROWTH_COEFFICIENT = 2;
+    private Node<K,V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        Node<K, V> node = table[hash(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key, value, null);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
+        table[hash(key)] = new Node<>(key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[hash(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            if (node.next != null) {
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        private Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int hash(K key) {
+        int keyHash = (key == null) ? 0
+                : (key.hashCode() > 0) ? key.hashCode()
+                : (-1) * key.hashCode();
```

Comment:
You can use Math.abs() instead of multiplying on (-1)

Reply:
Valid point, @DEz-1996, please pay attention